src/artifact/v3/header/header.cpp

```diff
@@ -235,6 +235,11 @@ ExpectedHeader Parse(io::Reader &reader, ParserConfig conf) {
 		current_index++;
 	}
 
+	if (tok.type == token::Type::Unrecognized) {
+		return expected::unexpected(parser_error::MakeError(
+			parser_error::Code::ParseError, "Unrecognized error while parsing the header"));
```

Comment:
~error~ token, right?

Reply:
Not really. The error is logged somewhere else and then we return "unrecognized" token. See [the code here](https://github.com/mendersoftware/mender/blob/121dbbd87c65344679c287b3f1e249a1372f3c4f/src/artifact/lexer.hpp#L57) and [the test here](https://github.com/mendersoftware/mender/pull/1690/commits/994f658e4ee139f26f52468f06908e7290d74996#diff-1321fc7c2ed177c9698ecd7318ddf2df02706110431451b56b2026072528dc0eR365).

I considered refactoring the lexer code to be able to return the error, but quickly realized that it was too much work for this bug.